modules/integration/tests-integration/lazy-loading-test/src/test/java/org/wso2/appserver/integration/lazy/loading/LazyLoadingBaseTest.java

```diff
@@ -0,0 +1,486 @@
+        } catch (AxisFault axisFault) {
+            String customErrorMessage = "AxisFault Exception  when creating WebAppAdminClient object. backend URL:" +
+                    backendURL + " Session Cookie: " + sessionCookie;
+            log.error(customErrorMessage, axisFault);
+            throw new LazyLoadingTestException(customErrorMessage, axisFault);
+        }
+        startTime = System.currentTimeMillis();
+        while (((time = (System.currentTimeMillis() - startTime)) < DEPLOYMENT_DELAY_IN_MILLISECONDS)) {
+            //List all applications
+            try {
+                appList = appAdminClient.listAllApplications();
+            } catch (ApplicationAdminExceptionException applicationAdminExceptionException) {
+                String customErrorMessage = "ApplicationAdminExceptionException Exception when when calling " +
+                        "listAllApplications() methods in WebAppAdminClient";
+                log.error(customErrorMessage, applicationAdminExceptionException);
+                throw new LazyLoadingTestException(customErrorMessage, applicationAdminExceptionException);
+            } catch (RemoteException remoteException) {
+                String customErrorMessage =
+                        "RemoteException Exception when calling listAllApplications() methods in WebAppAdminClient";
+                log.error(customErrorMessage, remoteException);
+                throw new LazyLoadingTestException(customErrorMessage, remoteException);
+            }
+            //Find given application is available in app list. if found  return isAppListed = true or else with false
+            if (Arrays.asList(appList).contains(appName)) {
+                isAppListed = true;
+                log.info(appName + " Carbon Application is listed in" + time + " millis");
+                break;
+            } else {
+                try {
+                    //Sleep 500 milliseconds until the next verification of Carbon application listing.
+                    Thread.sleep(500);
+                } catch (InterruptedException interruptedException) {
+                    String customErrorMessage = "InterruptedException occurs when sleeping 500 milliseconds and while" +
+                            " waiting for Carbon Application to  listed in ApplicationAdminClient.";
+                    log.warn(customErrorMessage, interruptedException);
+                }
+            }
+        }
+        return isAppListed;
+    }
+
+
+    /**
+     * Tenant admin user login functionality.
+     *
+     * @param domainKey -  Domain key of the tenant.
+     * @throws LazyLoadingTestException - Exception throws when  creating the AutomationContext and login() method of
+     *                                  LoginLogoutClient.java
+     */
+    protected void loginAsTenantAdmin(String domainKey) throws LazyLoadingTestException {
+        try {
+            AutomationContext automationContext =
```

Comment:
Maybe it's good to use proper constants upon string like "admin", "default", etc?